### DISCUSSION TITLE
Defining a custom Constant for Bigquery removes function call

### DISCUSSION THREAD

**Discussion by saschahofmann** [ibis-project/ibis#8213] [Q&A] [answered]
I am defining a custom constant that should compile to Bigquery's `CURRENT_DATE` function. I am in the process of updating to Ibis 7 but I don't really understand why this snippet that previously worked suddenly only compiles to `SELECT CURRENT_DATE` without calling the function.

```python
from ibis.expr.operations import Constant
import ibis.expr.datatypes as dt 
from ibis import bigquery
from ibis.backends.bigquery.compiler import BigQueryExprTranslator

compiles = BigQueryExprTranslator.compiles

class Today(Constant):
    dtype = dt.date


def today():
    return Today().to_expr()


@compiles(Today)
def _today(t, expr):
    return "CURRENT_DATE()"

bigquery.compile(today())
# returns 'SELECT\n  CURRENT_DATE AS `Today`'
# I expect 'SELECT\n  CURRENT_DATE() AS `Today`'

```

**Comment by saschahofmann** [comment]
@cpcloud Sorry for tagging you so shamelessly but I cannot figure this one out. It seems to be related to SQLGlot?

**Reply by jcrist** [reply, marked as answer]
I suspect this is due to `sqlglot` - in ibis 7.0 we started using sqlglot as part of the bigquery compilation chain. Does it make a difference though? From looking at the bigquery docs AFAICT both `CURRENT_DATE` and `CURRENT_DATE()` should be identical: https://cloud.google.com/bigquery/docs/reference/standard-sql/date_functions#current_date.

Since getting the current date is a general enough operation it makes sense to support it directly in ibis itself. I've pushed a PR up here: https://github.com/ibis-project/ibis/pull/8664.

**Reply by saschahofmann** [reply]
Interesting I didnt realise it would be the same and still work. 

And awesome thanks for the quick turnaround. I used to be able to kind of understand how ibis in the end compiled to SQL. Am I understand that now this is mostly done by SQLGlot. Could you maybe point me to the interface of ibis operations<sqlglot?

**Reply by saschahofmann** [reply]
Would love to select your response @jcrist as Answer but I cant figure out how :sweat_smile:

**Reply by cpcloud** [reply]
I marked it!

The discussion has to be a Q&A to be markable.

**Reply by saschahofmann** [reply]
Noted! Next question is posted as Q&A 😬
